### PR TITLE
feat(webapp): alternative solution for filter and order posts

### PR DESCRIPTION
**Pull request by Mogge**
### Issues

- None

### Todo

- [X] None

### CODE REVIEW

File: webapp/store/posts.spec.js
Comment (Tirokk) [inline webapp/store/posts.spec.js:1]: Do you find it a good idea to test for the default values of the store like I did it in my tests?
https://github.com/Ocelot-Social-Community/Ocelot-Social/blob/50768e27ecc1bf8074bb297a228741fe0542da62/webapp/store/posts.spec.js

**Review by Tirokk — APPROVED** [review]
Even if I'm not allowed to do so by our rules …

I give you my approve and thank you that you made the store lighter and much more clearly as well as the test slim. 🙏🏼 🚀🚀💫💫
I love it! 😍

Would be nice if you give a good hint here in the description to my PR #6341 and what it has todo with mine.
And I will do as well on mine if I close it.

PS: For transparence of our talk and because I feel I need to express my feelings to the team.

My PR came late. Sorry for that, but I was prepared to work on the weekend.

Like I said:
My PR was in a prereview state, because I was not satisfied with it and liked to go ahaead working on it. I got not the chance to better and improve my code myself and have a learning by doing or by having a pairing with you.
Because you throw my tests completetly I lost a half day of my work which makes me bad feelings as well.
I would've liked better if you have talked to me before you worked on my issue and code silently.
I'm sure we would have found a solution and you could have worked on my PR, instead of open a new one. That gave me the impression that I just do bullshit and that we are more in a kind of competition, even if you wrote on Discord that we are not.
Thanks for your excuse and for expressing your worries todo so in our talk.
And that we all will try todo better in the future.
I like always to learn and be very interested to have a good code base, that both is my main interest.

File: webapp/components/FilterMenu/EventsByFilter.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  components: {
+    FilterMenuSection,
+    LabeledButton,
+  },
+  computed: {
+    ...mapGetters({
+      postFilter: 'posts/filter',
+    }),
+    sectionTitle() {
+      // return $t('filter-menu.eventsEnded')
```
[the remark this answers is not in the document]
Comment (Elweyn) [reply webapp/components/FilterMenu/EventsByFilter.vue:44]:
```suggestion
```

File: webapp/store/posts.js [hunk 2]
```diff
@@ -81,6 +75,42 @@ export const mutations = {
...
+    const filter = clone(state.filter)
+    if (postType && !(filter.postType_in && filter.postType_in.includes(postType))) {
+      filter.postType_in = [postType]
+      if (postType === 'Event') {
+        filter.eventStart_gte = new Date()
+        state.order = 'eventStart_asc'
+      } else {
+        delete filter.eventStart_gte
+        state.order = 'createdAt_desc'
+      }
```
Comment (Elweyn) [inline webapp/store/posts.js:95]: This should be refactored as soon as we get a third postType!
Maybe a switch case with some methods!?

**Review by Elweyn — APPROVED** [review]

**Pull request merged**